src/Sylius/Bundle/ApiBundle/CommandHandler/Checkout/AddressOrderHandler.php

```diff
@@ -76,6 +77,19 @@ public function __invoke(AddressOrder $addressOrder): OrderInterface
             $order->setCustomer($this->provideCustomerByEmail($addressOrder->email));
         }
 
+        /** @var AddressInterface|null $billingAddress */
+        $billingAddress = $order->getBillingAddress();
+        /** @var AddressInterface|null $shippingAddress */
+        $shippingAddress = $order->getShippingAddress();
+
+        if ($billingAddress !== null) {
+            $this->manager->remove($billingAddress);
+        }
+
+        if ($shippingAddress !== null) {
+            $this->manager->remove($shippingAddress);
+        }
```

Comment:
Instead of removing, couldn't we update the existing addresses?

Reply:
Do you have any idea how? I try do it without using setters for all fields on adress object but it creates new object on database and leaves old. Using getters from new object and setters for old is less customizable IMO. All added new fields by whose who customize address will be not supported.

Reply:
Do you say, that if you set new values on the existing shipping/billing address (and flushes the manager), the new database entry is created? How is it possible? :) It should definitely not work this way, doctrine take care of the upgrades well for other entities